### PR TITLE
Use the partitioning module for Blivet-GUI in UI

### PR DESCRIPTION
**Pull request by poncovka**
Add DBus support for Blivet-GUI and use it in UI.
This is a functional version, but still in progress.

**DO NOT MERGE!**

### CODE REVIEW

File: pyanaconda/ui/gui/spokes/blivet_gui.py [hunk 1]
```diff
@@ -45,6 +49,66 @@
...
+
+    def get_actions(self):
+        """Get the current actions."""
+        # FIXME: Add DBus support for this.
+        # return self._storage_playground.devicetree.actions.find()
+        return []
+
+    def sort_actions(self):
+        """Prune and sort the actions."""
+        # FIXME: Shouldn't we always do this?
```
Comment (vojtechtrefny) [inline pyanaconda/ui/gui/spokes/blivet_gui.py]: We do this in custom spoke too -- https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/ui/gui/spokes/custom_storage.py#L1759

File: pyanaconda/ui/gui/spokes/blivet_gui.py [hunk 2]
```diff
@@ -45,6 +49,66 @@
...
+        # self._storage_playground.devicetree.actions.prune()
+        # self._storage_playground.devicetree.actions.sort()
+        pass
+
+    def set_new_swaps(self):
+        """Set new swaps."""
+        # FIXME: Add DBus support for this.
```
Comment (vojtechtrefny) [inline pyanaconda/ui/gui/spokes/blivet_gui.py]: You'll need this for custom spoke too -- https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/ui/gui/spokes/custom_storage.py#L226

File: pyanaconda/modules/storage/partitioning/blivet_interface.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+class BlivetPartitioningInterface(PartitioningInterface):
+    """DBus interface for the Blivet partitioning module."""
+
+    def SendRequest(self, data: List[Byte]) -> List[Byte]:
+        """Send a request to the storage handler.
+
+        :param data: a request data in bytes
+        :return: a reply data in bytes
+        """
+        server = AnacondaUtilsServer(self.implementation.handler, bytes(data))
```
Comment (vojtechtrefny) [inline pyanaconda/modules/storage/partitioning/blivet_interface.py]: I don't like the idea of creating the server class for every call, but I guess it's ok for now.
This definitely need some blivet-gui code changes in the future to allow better client-server communication customization.

File: pyanaconda/modules/storage/partitioning/blivet_interface.py [hunk 4]
```diff
@@ -17,11 +17,65 @@
...
+
+        elif unpickled_msg[0] == "param":
+            self._get_param(unpickled_msg)
+
+        elif unpickled_msg[0] == "method":
+            self._call_method(unpickled_msg)
+
+        elif unpickled_msg[0] == "next":
+            self._get_next(unpickled_msg)
+
```
Comment (jkonecny12) [inline pyanaconda/modules/storage/partitioning/blivet_interface.py]: Please remove the spaces between `if-else`.

File: pyanaconda/ui/gui/spokes/blivet_gui.py [hunk 5]
```diff
@@ -45,6 +49,66 @@
...
+    def initialize(self):
+        """Initialize the client."""
+        self._proxy = STORAGE.get_proxy(BLIVET_PARTITIONING)
+
+    def _send(self, data):
+        """Send a message to a server."""
+        self._result = bytes(self._proxy.SendRequest(data))
+
+    def _recv_msg(self):
+        """Receive a message from server."""
```
Comment (jkonecny12) [inline pyanaconda/ui/gui/spokes/blivet_gui.py]: `from *a* server`

**Comment by jkonecny12** [comment]
> Looks good to me. Thanks!

**Comment by jkonecny12** [comment]
Ups pardon...

File: pyanaconda/modules/storage/partitioning/blivet_interface.py (hunk 3 above)
Comment (poncovka) [reply pyanaconda/modules/storage/partitioning/blivet_interface.py]: Fixed.

File: pyanaconda/modules/storage/partitioning/blivet_interface.py (hunk 4 above)
Comment (poncovka) [reply pyanaconda/modules/storage/partitioning/blivet_interface.py]: Fixed.

File: pyanaconda/ui/gui/spokes/blivet_gui.py (hunk 5 above)
Comment (poncovka) [reply pyanaconda/ui/gui/spokes/blivet_gui.py]: Fixed.

**Pull request closed without merging**